### PR TITLE
task 26.1 done

### PR DESCRIPTION
**Pull request by Rommelua**

### CODE REVIEW

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 1]
```diff
@@ -4,6 +4,24 @@
...
+</div>
+<div>
+    <a href="${pageContext.request.contextPath}/products/add">Add product</a>
+</div>
+<div>
+    <a href="${pageContext.request.contextPath}/users/all">All users</a>
+</div>
+<div>
+    <a href="${pageContext.request.contextPath}/inject">Inject mock data to DB</a>
+</div>
```
Comment (vladkuzya) [inline src/main/webapp/WEB-INF/views/index.jsp]: Для удобства, и для правильного отображения списка (если я тебя правильно понял), я юзал 
```
<ul>
   <li>Element1</li>
   <li>Element2</li>
</ul>
```

Попробуй)

Comment (vladkuzya) [reply src/main/webapp/WEB-INF/views/index.jsp]: Ну а так, на вкус и цвет)

File: pom.xml [hunk 2]
```diff
@@ -31,6 +31,12 @@
             <version>3.1.0</version>
             <!--<scope>provided</scope> -->
```
Comment (Sofasmile) [inline pom.xml]: better remove commented code from repository

File: pom.xml [hunk 3]
```diff
@@ -31,6 +31,12 @@
...
         </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>jstl</artifactId>
+            <version>1.2</version>
+        </dependency>
+
```
Comment (Sofasmile) [inline pom.xml]: let's remove empty line

File: src/main/java/com/internet/shop/controller/RegistrationController.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class RegistrationController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/RegistrationController.java]: you should decide: to use the `final` with services or not (I recommend to use it)
and do it everywhere

File: src/main/java/com/internet/shop/controller/IndexController.java [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
+package com.internet.shop.controller;
 
 import java.io.IOException;
 import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+@WebServlet("/")
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/IndexController.java:10]: can we use one type of mapping? annotation or web.xml?

File: src/main/java/com/internet/shop/controller/RegistrationController.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String login = req.getParameter("login");
+        String password = req.getParameter("psw");
+        String passwordRep = req.getParameter("pswRep");
+
+        if (password.equals(passwordRep)) {
+            User user = new User(name, login, password);
+            userService.create(user);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/RegistrationController.java:38]: let's create a shopping cart for user

File: src/main/java/com/internet/shop/controller/cart/AddProductController.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/shopping-carts/products/add")
+public class AddProductController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/cart/AddProductController.java]:
```suggestion
public class AddProductToShoppingCartController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controller/cart/AddProductController.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private final ProductService productService
+            = (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
+        String productId = req.getParameter("id");
+        long id = Long.parseLong(productId);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/cart/AddProductController.java]: you can do as shown below (it's optional):
```suggestion
        Long id = Long.valueOf(productId);
```

File: src/main/java/com/internet/shop/controller/cart/DeleteProductController.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/shopping-carts/products/delete")
+public class DeleteProductController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/cart/DeleteProductController.java]: this is the next hw let's move it to the next pr

File: src/main/java/com/internet/shop/controller/user/DeleteUserController.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
+        long id = Long.parseLong(userId);
+        userService.delete(id);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/user/DeleteUserController.java]: let's remove users shopping cart here

File: pom.xml (hunk 2 above)
Comment (Rommelua) [reply pom.xml]: done

File: pom.xml (hunk 3 above)
Comment (Rommelua) [reply pom.xml]: done

File: src/main/java/com/internet/shop/controller/IndexController.java (hunk 5 above)
Comment (Rommelua) [reply src/main/java/com/internet/shop/controller/IndexController.java:10]: I have left only annotations

File: src/main/java/com/internet/shop/controller/RegistrationController.java (hunk 4 above)
Comment (Rommelua) [reply src/main/java/com/internet/shop/controller/RegistrationController.java]: done

File: src/main/java/com/internet/shop/controller/RegistrationController.java (hunk 6 above)
Comment (Rommelua) [reply src/main/java/com/internet/shop/controller/RegistrationController.java:38]: done

File: src/main/java/com/internet/shop/controller/cart/AddProductController.java (hunk 8 above)
Comment (Rommelua) [reply src/main/java/com/internet/shop/controller/cart/AddProductController.java]: Made no changes if you don't mind

File: src/main/java/com/internet/shop/controller/cart/DeleteProductController.java (hunk 9 above)
Comment (Rommelua) [reply src/main/java/com/internet/shop/controller/cart/DeleteProductController.java]: I really had not red next hw before creating this class. Removed.

File: src/main/java/com/internet/shop/controller/user/DeleteUserController.java (hunk 10 above)
Comment (Rommelua) [reply src/main/java/com/internet/shop/controller/user/DeleteUserController.java]: agree, done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**